cdk/domino_cdk/provisioners/eks/eks_iam_roles_for_k8s.py

```diff
@@ -0,0 +1,176 @@
+from typing import Dict, List
+
+import aws_cdk.aws_eks as eks
+import aws_cdk.aws_iam as iam
+from aws_cdk import core as cdk
+from aws_cdk.aws_s3 import Bucket
+
+# Permission groups
+
+s3_write_permissions = [
+    "s3:PutObject",
+    "s3:DeleteObject",
+    "s3:ListMultipartUploadParts",
+    "s3:AbortMultipartUpload",
+]
+
+s3_read_permissions = [
+    "s3:GetObject",
+]
+
+ecr_policies = {
+    "ecr_read_write_permissions": [
+        "ecr:PutImageTagMutability",
+        "ecr:StartImageScan",
+        "ecr:ListTagsForResource",
+        "ecr:UploadLayerPart",
+        "ecr:BatchDeleteImage",
+        "ecr:ListImages",
+        "ecr:DeleteRepository",
+        "ecr:CompleteLayerUpload",
+        "ecr:TagResource",
+        "ecr:DescribeRepositories",
+        "ecr:DeleteRepositoryPolicy",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:ReplicateImage",
+        "ecr:GetLifecyclePolicy",
+        "ecr:PutLifecyclePolicy",
+        "ecr:DescribeImageScanFindings",
+        "ecr:GetLifecyclePolicyPreview",
+        "ecr:CreateRepository",
+        "ecr:PutImageScanningConfiguration",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:DeleteLifecyclePolicy",
+        "ecr:PutImage",
+        "ecr:UntagResource",
+        "ecr:BatchGetImage",
+        "ecr:DescribeImages",
+        "ecr:StartLifecyclePolicyPreview",
+        "ecr:InitiateLayerUpload",
+        "ecr:GetRepositoryPolicy",
+    ]
+}
+
+s3_policies = {
+    "write_blobs_read_logs": {
+        "blobs": s3_read_permissions + s3_write_permissions,
+        "logs": s3_read_permissions,
+    },
+    "write_blobs": {
+        "blobs": s3_read_permissions + s3_write_permissions,
+    },
+    "read_logs": {
+        "logs": s3_read_permissions,
+    },
+    "read_blobs": {
+        "blobs": s3_read_permissions,
+    },
+}
+
+
+# Roles
+roles = {
+    "nucleus": [
+        "write_blobs_read_logs",
+    ],
+    "executor": [
+        "write_blobs",
+    ],
+    "builder": [
+        "ecr_read_write_permissions",
+    ],
+}
+
+
+class DominoEksK8sIamRolesProvisioner:
+    def __init__(
+        self,
+        scope: cdk.Construct,
+    ) -> None:
+        self.scope = scope
+
+    def print_children(self, construct):
+        print(construct.node.path, construct.to_string())
+        for c in construct.node.children:
+            self.print_children(c)
```

Comment:
Is this just a debug thing?

Reply:
Blindness.I though I have removed this debug thing.